src/BlazorWasmSdk/Tasks/ComputeBlazorBuildAssets.cs

```diff
@@ -97,7 +100,16 @@ public override bool Execute()
                     }
 
                     var destinationSubPath = candidate.GetMetadata("DestinationSubPath");
-                    if (string.IsNullOrEmpty(destinationSubPath))
+                    if (candidate.GetMetadata("FileName") == "dotnet" && candidate.GetMetadata("Extension") == ".js")
+                    {
+                        var dotNetVersion = MicrosoftNETCoreAppRefPackageVersion;
+                        var itemHash = FileHasher.GetFileHash(candidate.ItemSpec);
+                        var cacheBustedDotnetJSFileName = $"dotnet.{dotNetVersion}.{itemHash}";
+                        candidate.SetMetadata("RelativePath", $"_framework/{cacheBustedDotnetJSFileName}.js");
+                        candidate.SetMetadata("AssetTraitName", "BlazorWebAssemblyResource");
+                        candidate.SetMetadata("AssetTraitValue", "native");
```

Comment:
Could you clarify why we have to modify these asset trait metadata values? I expected to see code that changes the filename/path, but didn't expect to see other metadata get mutated. I'm sure there's a good reason but am curious to understand!

Reply:
We are doing this to help with another PR from the mono folks, we used to copy the dotnet.js file to the obj folder and rename it (because of limitations on old static web assets).

In addition to that, the mono folks are introducing support for debugging AoT apps at build time, at which point the dotnet.js file gets copied (by them) to the output folder and updated (for support debugging AoT with native libraries).

I met with Ankit and Larry yesterday to get a better understanding of it and determined that its simpler to avoid our own local copy and just look up the file and always create an asset that includes the version and the hash. That works for both cases, when AoT is being used and when its not

Reply:
OK. It's confusing when unrelated changes show up without explanation in a PR. Might be worth having a comment here, or future archaeologists/refactorers will be in for a tough time.

TBH there's so much of this build process that seems impossible to know just by reading the code that maybe I should just stop trying to reason about it. If it were me, I'd absolutely cover it in comments describing what external systems are relying on each of the moving parts here.

Reply:
> OK. It's confusing when unrelated changes show up without explanation in a PR. Might be worth having a comment here, or future archaeologists/refactorers will be in for a tough time.

I totally agree, sorry about that should've left a comment with some more context. I've updated the PR description with Javier's comment above to provide reasoning for these additional changes.